docs/authoring.md

```diff
@@ -61,16 +63,44 @@ components:
         id:
           type: integer
 ```
+
 The generator creates two files `peoplePayload.txt` and `people.txt` with the following content:
 ```
 Schema name is 'peoplePayload' and properties are:
 - people
 ```
+
 and
 ```
 Schema name is 'people' and properties are:
 - id
 ```
+
+### React
+
+The above way of rendering **file templates** works for both `nunjucks` and `react` render engine, but `react` also has another, more generic way to render multiple files. It is enough to return an array of `File` components in the rendering component. Check example:
+
+```tsx
+export default function({ asyncapi }) {
+  if (!asyncapi.components().hasSchemas()) {
+    return null;
+  }
+
+  const schemas = asyncapi.components().schemas();
+  return Object.entries(schemas).map(([schemaName, schema]) => {
+    return (
+      <File name={`${schemaName}.html`}>
+        <SchemaFile schemaName={schemaName} schema={schema} />
+      </File>
+    );
+  });
+}
+
+function SchemaFile({ schemaName, schema }) {
+  // implementation...
+}
+```
```

Comment:
I think this is too large of an example, what about: 

```suggestion
```tsx
export default function({ asyncapi }) {
  return [
    <File name={`file1.html`}>//Content</File>,
    <File name={`file2.html`}>//Content</File>
  ]
}
```
```